#! /bin/sh /usr/share/dpatch/dpatch-run
## 33increase_max_winmsg_renditions.dpatch
## by Dustin Kirkland <kirkland@canonical.com>
##
## DP: Increase MAX_WINMSG_REND constant from 16 to 80
## DP: Allows for up to 64 formatting changes on a status line
## DP: https://bugs.launchpad.net/ubuntu/+source/screen/+bug/315871

@DPATCH@
diff -uprN screen-4.0.3.orig/screen.c screen-4.0.3/screen.c
--- screen-4.0.3.orig/screen.c	2003-09-08 09:26:41.000000000 -0500
+++ screen-4.0.3/screen.c	2009-01-10 12:06:34.698560196 -0600
@@ -2026,7 +2026,7 @@ static const char months[] = "JanFebMarA
 #endif
 
 static char winmsg_buf[MAXSTR];
-#define MAX_WINMSG_REND 16	/* rendition changes */
+#define MAX_WINMSG_REND 80	/* rendition changes */
 static int winmsg_rend[MAX_WINMSG_REND];
 static int winmsg_rendpos[MAX_WINMSG_REND];
 static int winmsg_numrend;
